algebird-core/src/main/scala/com/twitter/algebird/Field.scala

```diff
@@ -54,6 +54,14 @@ object FloatField extends Field[Float] {
     assertNotZero(r)
     l / r
   }
+  override def sum(t: TraversableOnce[Float]): Float = {
+    var sum = 0.0f
+    t.foreach(sum += _)
```

Comment:
OK thanks, didn't really have an intuition for what the cost of method dispatch vs boxing (object allocation?) is. But yeah, seems like here we might as well use a while anyway.